group-coordinator/src/main/java/org/apache/kafka/coordinator/group/generic/GenericGroup.java

```diff
@@ -849,6 +853,43 @@ public void validateOffsetFetch(
         }
     }
 
+    /**
+     * Validates the OffsetDelete request.
+     */
+    @Override
+    public void validateOffsetDelete() throws ApiException {
+        if (isInState(DEAD)) {
+            throw new GroupIdNotFoundException(String.format("Group %s is in dead state.", groupId));
+        } else if (isInState(STABLE)
```

Comment:
don't we also need to check whether the stable group is using the ConsumerProtocol.PROTOCOL_TYPE?
from
```
                case PreparingRebalance | CompletingRebalance | Stable if group.isConsumerGroup =>
```